File: LeetCode/201-300/Problem216_CombinationSumIII/CombinationSumIII.py

class Solution:
    def combinationSum3(self, k: int, n: int) -> List[List[int]]:
        result = []

        def backtrack(start: int, current: List[int], target: int):
            if len(current) == k:
                if target == 0:
                    result.append(current[:])
                return
            
            for i in range(start, 10):
                if i > target:
                    break  
                current.append(i)
                backtrack(i + 1, current, target - i)
                current.pop() 

        backtrack(1, [], n)
        return result

File: LeetCode/201-300/Problem216_CombinationSumIII/CombinationSumIII.cpp
class Solution {
public:
    vector<vector<int>> combinationSum3(int k, int n) {
        vector<vector<int>> result;
        vector<int> current;
        backtrack(1, k, n, current, result);
        return result;
    }

private:
    void backtrack(int start, int k, int target, vector<int>& current, vector<vector<int>>& result) {
        if (current.size() == k) {
            if (target == 0) {
                result.push_back(current);
            }
            return;
        }

        for (int i = start; i <= 9; ++i) {
            if (i > target) break; 
            current.push_back(i);
            backtrack(i + 1, k, target - i, current, result);
            current.pop_back(); 
        }
    }
};
